modules/openapi-generator/src/main/resources/dart2/api_client.mustache

```diff
@@ -191,18 +192,17 @@ class ApiClient {
     try {
       switch (targetType) {
         case 'String':
```

Comment:
when this will not be 'String' type after passing this 'case' condition?

Reply:
Unsure what you mean here... The switch is checking `targetType` which may be `String`, `int`, `double`, and all other defined models.

Reply:
Maybe I understood... do you mean to ask "why the code that checks whether `value is String` in Dart? If so, then 2 reasons:
1) to be similar to the treatment with `int`, `double` and `bool`.
2) in case the API changes or misbehaves, the client at least doesn't crash.